### PR TITLE
Branch update DG-Draft

### LINKED ISSUE #108: Update Developer Guide (DoNext Command)

### PR DESCRIPTION
**Pull request by eyelessrhyme7**
Added implementation details for do_next command, its relevant classes and methods. Yet to add sequence diagrams. Will add them in subsequent iterations.

closes #108

### CODE REVIEW

**Review by yarnmengnus — APPROVED** [review]
lgtm ty for the good work!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -155,6 +155,88 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+### Usage <a name="usage"></a>
+To use the `DoNext` in the ProfPlan application, you can execute the following steps:
+
+1. Enter the "do_next" command.
+
+2. The `DoNext` will handle the task recommendation generation.
+
+3. The application will provide feedback, and output the task generated in the message window.
+
```
Comment (kiatkat) [inline docs/DeveloperGuide.md:239]: Wow!

**Review by kiatkat — APPROVED** [review]
LGTM.

**Review by newway1814 — APPROVED** [review]
Great Job!

Follows the structure!

**Pull request merged**